client/grpc_query.go

```diff
@@ -56,8 +56,8 @@ func (ctx Context) Invoke(grpcCtx gocontext.Context, method string, req, reply i
 	// As a result, we direct them to the ABCI flow where they get syncronized.
 	_, isSimulationRequest := req.(*tx.SimulateRequest)
 	isTendermintQuery := strings.Contains(method, "tendermint")
-
-	isGRPCAllowed := !isTendermintQuery && !isSimulationRequest
+	grpcConcurrentEnabled := ctx.GRPCConcurrency
```

Comment:
Please add a table-driven test `TestGRPCQuery_Bank_Service_GRPC_Concurrency` where this flag is true and where it is false. Here is a [similar example](https://github.com/osmosis-labs/cosmos-sdk/blob/f3613c8c7ab6a26ba35613a7c205e49e9b07ebcd/client/grpc_query_test.go#L69)

Reply:
In my latest commit, I basically added `clientCtx.GRPCConcurrency = true` to all the table driven tests that were failing in order to get them to pass. I am not quite understanding 
a) why they were failing in the first place due to this setting and
b) what exactly I should be expanding on in terms of tests with and without the flag (i.e. I dont understand what the expected outputs should actually be)

If you could provide some fidelity it would be much appreciated

Reply:
a) The tests were failing because they were set up assuming that GRPC is always enabled. Since we added a flag to disable them in some cases, some of them started failing due to the incorrect setup.

b) Our goal is to test the newly added `grpcConcurrentEnabled` flag. We would have to test all possible combinations of inputs with the newly introduced flag for maximum test coverage. However, I think it is sufficient to at least test the true and false values of `grpcConcurrentEnabled`  since this change is a low risk / low impact.

To achieve the suggested minimum, you can repeat [this test](https://github.com/osmosis-labs/cosmos-sdk/blob/f69c198375023861132ed5b3a75374940000f46a/client/grpc_query_test.go#L59-L67) as a table-driven test in a separate fixture, replacing the `Echo(...)` function will [`Invoke(ctx, "/testdata.Query/Echo", ...)`](https://github.com/osmosis-labs/cosmos-sdk/blob/f69c198375023861132ed5b3a75374940000f46a/testutil/testdata/query.pb.go#L360). `Echo(...)` is just a wrapper around `Invoke(...)` and we want to make sure that we are unit testing `Invoke(...)` directly